opencensus/metrics_quickstart/main.go

```diff
@@ -63,23 +63,25 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
-	// Flush must be called before main() exits to ensure metrics are recorded.
-	defer exporter.Flush()
 
 	if err := exporter.StartMetricsExporter(); err != nil {
 		log.Fatalf("Error starting metric exporter: %v", err)
 	}
-	defer exporter.StopMetricsExporter()
 	// [END monitoring_opencensus_setup_exporter]
 
-	// Record 100 fake latency values between 0 and 5 seconds.
-	for i := 0; i < 100; i++ {
+	// Record 50 fake latency values between 0 and 5 seconds.
+	for i := 0; i < 50; i++ {
 		ms := float64(5*time.Second/time.Millisecond) * rand.Float64()
 		fmt.Printf("Latency %d: %f\n", i, ms)
 		stats.Record(ctx, latencyMs.M(ms))
 		time.Sleep(1 * time.Second)
 	}
 
+	// Metrics are reported every 60 seconds to the backend. If number of
+	// iteration is changed above or if the interval between iteration is changed
+	// then adjust the sleep below to allow for all iterations to be exported.
+	time.Sleep(15 * time.Second)
```

Comment:
Call `exporter.Flush()` to force a flush right now (and edit the comment to say why we're calling `Flush`)?

Before, we did `defer exporter.Flush()` -- did that not work? If not, it sounds like a bug.

https://godoc.org/contrib.go.opencensus.io/exporter/stackdriver#Exporter.Flush

Reply:
Flush() was creating a confusion. Flush() only takes care of flushing the internal buffer of the exporter. It doesn't read all the metrics again and export them. For example,
At time T0, Exporter reads all (let say 10) the metrics and queues them in a buffer to export at a later time.
At time T1, application update few (let say 5) metrics.
At time T2, application terminates, calling Flush.
Flush will only make sure that internal buffer is flushed. It will not read the metrics again.
Some may expect that Flush read all metrics again and then export them, which is going to be tricky as discussed in other comments in this PR.
Alternatively, I can restore the Flush() in the example, and clearly put a comment as follows
```
// The exporter reads metrics periodically and queues them to an internal buffer.
// Flush() will ensure that all metric queued in the buffer are exported to the backend. 
```

Reply:
That sounds like a bug in `Flush`? The docs say to call it to avoid losing data, but you can still lose data. Can we update `Flush` to read all of the data?

Earlier, you said
> However, it needs to make sure that it waits for enough time since last report so that the backend doesn't reject the report (due to it being too soon).

Sorry I'm so confused here. If the backend can reject it if its sent data too soon, it seems like you should never call `Flush`?

Reply:
Note that platforms such as Cloud Run create a need for a solid flush() method and if samples don't include it there will be:

* A need for platform-specific metrics samples
* An increased number of developers losing metrics because they never look at the platform-specific sample

This comment almost exceeds scope of the PR, and I appreciate the goal of keeping this simple.